src/perf.js

```diff
@@ -0,0 +1,12 @@
+export const memoize = function (fn) {
```

Comment:
Happy to make a change here if you think it is necessary, but let me explain my thinking.  If you look at the `memoize` function that is included with Lodash, there isn't much to it.  If a `resolver` is included, all arguments _will_ be passed (this is not the case with Underscore's `memoize`), which makes using `Lodash#memoize` a possibility.

However, the `resolver` key-generation will consist of basically all of the code in this PR's `memoize`, plus overhead of an additional `fn.apply`.  Since the motivation for this PR is to improve performance, and because this `memoize` is in the hot code path, I wanted to reduce any unnecessary overhead.